src/DevChatter.Bot.Infra.Twitch/TwitchChatClient.cs

```diff
@@ -32,21 +33,41 @@ private void ChatCommandReceived(object sender, OnChatCommandReceivedArgs e)
             OnCommandReceived?.Invoke(this, e.ToCommandReceivedEventArgs());
         }
 
+        public async Task Connect()
+        {
+            _twitchClient.OnConnected += TwitchClientConnected;
+            _twitchClient.Connect();
+
+            await _connectionCompletionTask.Task;
+        }
+
         private void TwitchClientConnected(object sender, OnConnectedArgs onConnectedArgs)
         {
+            _twitchClient.OnConnected -= TwitchClientConnected;
+
             _isReady = true;
             _connectionCompletionTask.SetResult(true);
+            _disconnectionCompletionTask = new TaskCompletionSource<bool>();
             _twitchClient.SendMessage("Hello World! The bot has arrived!");
         }
 
-        public async Task Connect()
+        public async Task Disconnect()
         {
-            _twitchClient.Connect();
-            _twitchClient.OnConnected += TwitchClientConnected;
+            _twitchClient.OnDisconnected += TwitchClientOnOnDisconnected;
+            _twitchClient.Disconnect();
 
             await _connectionCompletionTask.Task;
```

Comment:
Shouldn't this be awaiting `_disconnectionCompletionTask .Task`?

Reply:
Yes. I changed it in the latest commit.